### ISSUE TITLE
[JS/TS] Desktop gets confused with `` strings

### ISSUE THREAD

**Issue by ImRodry** [desktop/desktop#12489]
### Describe the bug
When your file has \`\` strings at any point, GitHub desktop will treat every line of text from that point onwards as a string in the color highlighting (excluding ${}. It also stops sometimes but I haven't been able to identify why or when)

### Version & OS
Version 2.9.0 
Windows 10 version 21H1 build 19043.1055

### Steps to reproduce the behavior

1. Go to a javascript or typescript repository/file
2. Insert a string using \`\` instead of ""
3. Go to GitHub desktop and observe the diff before and after that string

### Expected behavior
The string is properly highlighted, not messing with anything else

### Actual behavior
Every line of text from that point onwards is highlighted as a string

### Screenshots
![image](https://user-images.githubusercontent.com/38259440/122796973-e141d180-d2b6-11eb-9c59-2969724fae2c.png)

### Logs
Don't think it's applicable.

**Comment by tidy-dev** [comment]
@ImRodry Thanks for the report. I could only reproduce this when there was the `?` operator also used inside the \`\` string like  \`{someobject?.someproperty}\`. Unfortunately I am not sure this will be easy fix as we use a library to do this parsing. But, marking it as bug to be placed on the backlog.

**Comment by ImRodry** [comment]
Yeah that seems to be the case. I remember a few versions ago this didn't use to happen so maybe something updated on that library that broke this? Could report it to them possibly

**Comment by ImRodry** [comment]
Just adding another example without using ? where this bug is reproduceable. Not sure what happened here though
![image](https://user-images.githubusercontent.com/38259440/123954539-6e181980-d9a0-11eb-9d79-6a4c22995879.png)

**Comment by say25** [comment]
I think this is going to depend on upgrading CodeMirror.

**Comment by say25** [comment]
I believe this was fixed as part of #11950. Please reopen if this is still an issue 
![image](https://user-images.githubusercontent.com/7467062/137351571-64356517-25a1-4268-995f-7b4d6caad644.png)

**Comment by ImRodry** [comment]
@say25 doesn't seem to be fixed on my end on v2.9.4. Here's a screenshot from https://github.com/discordjs/discord.js/blob/4dff279a6f88612db9283851fba8850a2d6c65fe/typings/index.d.ts#L3349-L3365
(this behavior can also be found on the GitHub website btw)
![image](https://user-images.githubusercontent.com/38259440/137793025-d36d1253-e402-427f-92bb-318ea1e7a657.png)
